redisearch/pool.go

```diff
@@ -65,3 +65,18 @@ func (p *MultiHostPool) Get() redis.Conn {
 	return pool.Get()
 
 }
+
+func (p *MultiHostPool) Close() (err error) {
+	p.Lock()
+	defer p.Unlock()
+	for _, host := range p.hosts {
+		pool, found := p.pools[host]
+		if found {
+			err = pool.Close()
+			if err != nil {
+				return
```

Comment:
if any pool close() errors out, you want to return and not proceed with the rest pools? and why not just ranging over `p.pools` instead? seems more straight forward to me